src/local_scheduler/local_scheduler.cc

```diff
@@ -124,9 +125,12 @@ void kill_worker(LocalSchedulerClient *worker, bool cleanup) {
 
   /* Clean up the task in progress. */
   if (worker->task_in_progress) {
-    /* Return the resources that the worker was using. */
-    TaskSpec *spec = Task_task_spec(worker->task_in_progress);
-    update_dynamic_resources(state, spec, true);
+    if (!worker->is_blocked) {
```

Comment:
Yup, we were sometimes going over `static_resources` (a fatal error) if a worker was blocked while freeing the local scheduler state.